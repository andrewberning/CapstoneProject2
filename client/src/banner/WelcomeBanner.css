.WelcomeBanner {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: var(--light-clr);
}

.WelcomeBanner .banner-container {
  text-align: center;
  margin-block: 50px;
  color: var(--secondary-clr);
}