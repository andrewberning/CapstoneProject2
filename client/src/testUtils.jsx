import React from "react";
import UserContext from "./auth/UserContext";
import { vi } from 'vitest';

const demoUser = {
  username: "testuser",
  first_name: "Test",
  last_name: "User",
  email: "testuser123@test.net",
};

// Mock functions
const mockSetCurrentUser = vi.fn();
const mockLogin = vi.fn();
const mockLogout = vi.fn();
const mockSignup = vi.fn();
const mockAddToCart = vi.fn();
const mockHasAddedToCart = vi.fn();

const UserProvider = ({ 
  children,
  currentUser = demoUser,
  setCurrentUser = mockSetCurrentUser,
  login = mockLogin,
  logout = mockLogout,
  signup = mockSignup,
  addToCart = mockAddToCart,
  hasAddedToCart = mockHasAddedToCart,
  cartItems = [],
  setCartItems = () => {},
  categories = [],
  infoLoaded = true,
  totalAmount = 0,
  totalItems = 0,
}) => (
  <UserContext.Provider value={{ 
    currentUser,
    setCurrentUser,
    login,
    logout,
    signup,
    addToCart,
    hasAddedToCart,
    cartItems,
    setCartItems,
    categories,
    infoLoaded,
    totalAmount,
    totalItems,
  }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider, mockSetCurrentUser, mockLogin, mockLogout, mockSignup, mockAddToCart, mockHasAddedToCart };
