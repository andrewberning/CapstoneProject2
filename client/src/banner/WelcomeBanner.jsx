import { useContext } from "react";
import UserContext from "../auth/UserContext";
import "./WelcomeBanner.css";

export default function WelcomeBanner() {
  const { currentUser } = useContext(UserContext);

  return (
    <section className="WelcomeBanner d-flex flex-column justify-content-center align-items-center">
      <div className="text-center ">
        {currentUser
          ? <h2>Welcome Back, {currentUser.firstName}!</h2>
          : <h2>Welcome to Shoply</h2>
        }
        <p className="lead">Everything in one, convenient place.</p>
      </div>
    </section>
  );
}