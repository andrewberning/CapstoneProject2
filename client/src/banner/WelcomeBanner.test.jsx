import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import WelcomeBanner from "./WelcomeBanner";
import { UserProvider } from "../testUtils";

describe('CategoriesBanner component', () => {
  it('renders without crashing', () => {
    render(
      <MemoryRouter>
        <UserProvider >
          <WelcomeBanner />
        </UserProvider>
      </MemoryRouter>
    );
  });

  it("matches snapshot", function () {
    const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <WelcomeBanner />
        </UserProvider>
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});