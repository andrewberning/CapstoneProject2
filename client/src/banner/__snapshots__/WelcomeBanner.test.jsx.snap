// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CategoriesBanner component > matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="WelcomeBanner d-flex justify-content-center align-items-center"
  >
    <div
      class="text-center "
    >
      <h2>
        Welcome to Shoply !
      </h2>
      <p
        class="lead"
      >
        Everything in one, convenient place.
      </p>
    </div>
  </section>
</DocumentFragment>
`;
